webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/uploading/UploadService.scala

```diff
@@ -296,6 +319,14 @@ class UploadService @Inject()(dataSourceRepository: DataSourceRepository,
       datasetNeedsConversion = uploadInformation.needsConversion.getOrElse(false)
       uploadDir = uploadDirectory(dataSourceId.organizationId, uploadId)
       unpackToDir = dataSourceDirFor(dataSourceId, datasetNeedsConversion)
+      totalFileSizeInBytesOpt <- runningUploadMetadataStore.find(redisKeyForTotalFileSizeInBytes(uploadId))
+      _ <- Fox.runOptional(totalFileSizeInBytesOpt) { maxFileSize =>
+        tryo(FileUtils.sizeOfDirectoryAsBigInteger(uploadDir.toFile).longValue).map(actualFileSize =>
```

Comment:
did you double-check that this returns the same value as the frontend-side check? (both for archives and multi-file uploads)?

Reply:
How big are our chunks? Could this become slow if it runs for every chunk?